cirq-core/cirq/transformers/measurement_transformers.py

```diff
@@ -104,7 +104,7 @@ def defer(op: 'cirq.Operation', _) -> 'cirq.OP_TREE':
             key = value.MeasurementKey.parse_serialized(gate.key)
             targets = [_MeasurementQid(key, q) for q in op.qubits]
             measurement_qubits[key] = targets
-            cxs = [ops.CX(q, target) for q, target in zip(op.qubits, targets)]
+            cxs = [_cx(q.dimension).on(q, target) for q, target in zip(op.qubits, targets)]
```

Comment:
Optional suggestion: It's a little awkward that we have to pass dimension and then qudits since the latter uniquely determine the former. You could just say

```python
    cxs = [_mod_add(q, t) for q, t in zip(op.qubits, targets)]
```

which hides dimension entirely, if you redefine `_cx` to

```python
def _mod_add(source: Qid, target: Qid) -> Operation:
    assert source.dimension == target.dimension
    if source.dimension == 2:
        return ops.CX.on(source, target)
    return _Add(source.dimension).on(source, target)
```

Reply:
Good suggestion, thanks!